crates/wash-lib/src/build/mod.rs

```diff
@@ -54,9 +54,14 @@ pub async fn load_lock_file(dir: impl AsRef<Path>) -> Result<LockFile> {
             .await
             .context("failed to load lock file")
     } else {
-        LockFile::new_with_path([], lock_file_path)
+        let mut lock_file = LockFile::new_with_path([], lock_file_path)
             .await
-            .context("failed to create lock file")
+            .context("failed to create lock file")?;
+        lock_file
```

Comment:
For knowledge purposes, the main issue that is causing this is that we create an empty file and then at some point try to load the lock file again (so I don't quite know where that is happening). You can see that this is what happens if you look at the docs for `new_with_path`.

Reply:
Yep, I followed down the path to see what was happening in `wasm_pkg_core` and came to the conclusion that our best here would be to persist the `LockFile` before we pass it back.